### DISCUSSION TITLE
Some sources equal zero in energy dashboard

### DISCUSSION THREAD

**Discussion by abertram** [nathanmarlor/foxess_modbus#674] [Support]
Hi,

I am using HA since a few weeks. I own a H3 10.0 which is connected to a USR TCP232 410s since a few days.

During initial setup with an automatic configuration of the energy dashboard "Grid Consumption Total" and "Feed-in Total" were always zero in energy dashboard. I assume it is due to no changes of the corresponding sensor value.

After a few hours I did uninstall the whole plugin and did install it again. "Grid Consumption Total" and "Feed-in Total" were changing and values were appearing in the energy dashboard. But at the same time "PV1 Power Total" and "PV2 Power Total" were gone like "Grid Consumption Total" and "Feed-in Total" before. I tried to remove and reinstall the plugin again. But it didn't help.

Energy dashboard configuration:

![image](https://github.com/user-attachments/assets/9edb5a5d-16f4-48cb-a843-fc1e70311aaa)

Energy dashboard:

![image](https://github.com/user-attachments/assets/72b5e397-7165-44a5-aba3-87f84120ed0d)

![image](https://github.com/user-attachments/assets/788b73b5-3c05-42b9-b98e-ed2e46cac57a)

![image](https://github.com/user-attachments/assets/210f006a-eeaa-4ffc-9d5c-2622bd97cfc6)

Could you please help me to solve the issue?

Best,
Alex

**Comment by hooktheboss** [comment]
Have you checked the readings at settings/devices/foxess modbus:
![grafik](https://github.com/user-attachments/assets/edbced48-8a65-4880-98fd-c98172d6b2df)
Click! Have you all sensor populated (at least after a while like this)?:
![grafik](https://github.com/user-attachments/assets/3c67d31d-df94-4605-b24b-866d3750df5b)

**Reply by abertram** [reply]
Yes, I did and it looks ok.

![image](https://github.com/user-attachments/assets/b29f0898-97e8-4d1c-a3ed-3885219623b2)

And yes almost all values are populated except PV1 and PV2 values.

![image](https://github.com/user-attachments/assets/ee41d59d-d050-485f-b39e-326062b09db8)

A few days ago it was working and then it got stuck.

![image](https://github.com/user-attachments/assets/1ba36234-fbe4-4f52-a197-72dbd1e7e597)

**Comment by hooktheboss** [comment]
Only difference I see is the additional (FoxESSInverter) in my values titles, but I don't recall where I exactly got that from - I assume in setup process.
I also recall, that in the begin for days I got NO PV1 Power Total, just Solar Generation Total. Later on this value kicked in (days later):
![grafik](https://github.com/user-attachments/assets/78d57445-66f8-4234-ab05-9a37ed34addf)

**Reply by abertram** [reply]
Ok, thank you!

**Reply by FozzieUK** [reply]
@abertram 
Some things to look at -

Go to Settings, Devices & Services, and click on '1 Device' in Foxess Modbus - find 'PV1 Power Total' in the list and click on it - then click on the cog icon in the top right hand corner - can you past the image here (I particularly want to see the Entity ID 5th line down)

Go into Developer Tools. click 'STATISTICS' at the top, search for 'PV1 Power Total' - does it say 'No Issue' in the 'issue' column, if there is an issue it will be accompanied by a 'FIX ISSUE' button at the end of the line - click that and past the dialogue info here.

**Reply by abertram** [reply]
Hi @FozzieUK,

looks like there is no issue. :-(

![image](https://github.com/user-attachments/assets/af091b4c-2832-4337-87e3-27ee3b3cc43a)

![image](https://github.com/user-attachments/assets/6d1fe74d-b511-4966-980b-3c7e705a3def)

Best,
Alex

**Reply by FozzieUK** [reply]
Ok, thanks

Can I just check something that wasn’t clear from one of your earlier posts - do the PV1 and PV2 power sensors (not the Totals) show history graphs when you click on them?

**Reply by abertram** [reply]
Yes, they do.

![image](https://github.com/user-attachments/assets/5bb74a39-5e1e-4990-9338-d268a041815b)

![image](https://github.com/user-attachments/assets/f3f84bcc-5853-4733-ad1d-ad1e61b47e39)

**Reply by FozzieUK** [reply]
Ok, thats great - just worried that they were showing 0 on a previous post.

So another thing to try.

Go to the Energy dashboard and if there is no solar production showing, click the 3 dots in the corner of the Energy Dashboard, and choose ‘Energy Configuration’ - in the ‘Solar Panel’ settings click the bin icon next to PV1 Power Total and PV2 Power Total to delete them.

Then can you restart your home assistant (whilst PV is being generated ), leave it a couple of hours and then go back to the Energy dashboard, again click on the 3 dots in the corner, ‘Energy Configuration’ and under Solar Panels, click to add solar production - type ‘PV1 Power Total’ and select it, choose not to forecast solar and ‘Save’,  repeat that for ‘PV2 Power Total’.

Leave it for at least 2 hours whilst there is solar production and go back to the energy dashboard to check it is showing correctly.

**Reply by abertram** [reply]
Hi @FozzieUK,

I assume they were zero due making screenshots at night.

Deleting and adding sensors didn't work. :-(

In the mean time I also tried to uninstall the whole integration, to delete the sensors and to reset the statistics. Didn't help as well. :-(

Are there any logs I can check for more information?

Best,
Alex

**Reply by abertram** [reply]
Hi @FozzieUK,

just checked the energy dashboard again and I see the values changing. I don't know why to be honest. I will observe and will keep you updated.

Thank you!

Best,
Alex

**Reply by FozzieUK** [reply]
Home assistant needs a couple of hours (minimum) for statistics to collect before it starts to display anything - i’m guessing it just needed a bit of time to gather data - i’ll keep my fingers crossed 😄
